packages/autotest/src/autotest/GradingJob.ts

```diff
@@ -100,21 +100,24 @@ export class GradingJob {
 
         const stdio = fs.createWriteStream(this.path + "/staff/stdio.txt");
         const stream = await container.attach({stream: true, stdout: true, stderr: true});
-        stream.pipe(stdio);
+        container.modem.demuxStream(stream, stdio, stdio);
 
         const exitCode = await GradingJob.runContainer(container, maxExecTime);
 
         const out = this.record.output;
-        if (exitCode === -1) {
-            out.report.feedback = "Container did not complete for `" + this.input.delivId + "` in the allotted time.";
-            out.state = ContainerState.TIMEOUT;
-        } else {
-            try {
-                const shouldPostback: boolean = exitCode !== 0;
-                out.report = await fs.readJson(this.path + "/staff/report.json");
-                out.postbackOnComplete = shouldPostback;
-                out.state = ContainerState.SUCCESS;
-            } catch (err) {
+
+        try {
+            // Shouldn't always postback on exitcode -1, as could have timed out _after_ report was made
```

Comment:
This looks like a precautionary step that we may consider in the root Classy project. 

@braxtonhall I am curious how a container could exit with an exitcode -1 after the report was made. Was this due to a problem with the internal container grading logic timing out or erring instead of any problems with the student code?

Reply:
I'm actually not comfortable with this going upstream yet as it didn't come out the way we wanted, and are still thinking about it. @LucasZamprogno and I will probably chew on it again after exams.

Currently, the grading container is capable of making a report (for important things like grades) and then continue running additional tasks like collecting repo/test data which could possibly time out.

Reply:
Sounds good @braxtonhall. I have to sync the fork up with the root project, so if I can pull in `ubccpsc310` changes without this file, will it work? I don't think this logic is entangled with any other logic, but I just wanted to double-check. 

I can make a note not to bring this file into the root Classy project during the merge, as it is still under development and requires testing.

Any concerns with that approach?

Reply:
Yeah it will work if you omit this file, although the change a few lines up is a nice QOL bugfix
```
container.modem.demuxStream(stream, stdio, stdio);
```

Reply:
Yeah, this does look good. I want it, but it looks like this file will have to wait until we can narrow the scope down. I will make a note of that demux stream change though. Thanks!